cmd/gaia/init/init.go

```diff
@@ -159,149 +70,87 @@ func InitCmd(ctx *server.Context, cdc *codec.Codec, appInit server.AppInit) *cob
+		}
+		pk := readOrCreatePrivValidator(privValFile)
+		fmt.Fprintf(os.Stderr,"validator public key: %s\n", sdk.MustBech32ifyConsPub(pk))
+		return
 	}
-	chainID = initConfig.ChainID
+
 
 	genFile := config.GenesisFile()
-	if !initConfig.Overwrite && common.FileExists(genFile) {
+	if !overwriteGenesis && common.FileExists(genFile) {
 		err = fmt.Errorf("genesis.json file already exists: %v", genFile)
 		return
 	}
 
 	// process genesis transactions, or otherwise create one for defaults
-	var appGenTxs []json.RawMessage
-	var validators []types.GenesisValidator
+	var appGenTxs []auth.StdTx
 	var persistentPeers string
+	var genTxs []json.RawMessage
+	var appState json.RawMessage
+	var validators []types.GenesisValidator
 
-	if initConfig.GenTxs {
-		validators, appGenTxs, persistentPeers, err = processGenTxs(initConfig.GenTxsDir, cdc)
+	if withGenTxs {
+		_, appGenTxs, persistentPeers, err = app.ProcessStdTxs(moniker, genTxsDir, cdc)
 		if err != nil {
 			return
 		}
+		genTxs = make([]json.RawMessage, len(appGenTxs))
 		config.P2P.PersistentPeers = persistentPeers
-		configFilePath := filepath.Join(config.RootDir, "config", "config.toml")
-		cfg.WriteConfigFile(configFilePath, config)
-	} else {
-		genTxConfig := servercfg.GenTx{
-			viper.GetString(server.FlagName),
-			viper.GetString(server.FlagClientHome),
-			viper.GetBool(server.FlagOWK),
-			"127.0.0.1",
-		}
-
-		// Write updated config with moniker
-		config.Moniker = genTxConfig.Name
-		configFilePath := filepath.Join(config.RootDir, "config", "config.toml")
-		cfg.WriteConfigFile(configFilePath, config)
-		appGenTx, am, validator, err := appInit.AppGenTx(cdc, pubKey, genTxConfig)
-		appMessage = am
-		if err != nil {
-			return "", "", nil, err
-		}
-		validators = []types.GenesisValidator{validator}
-		appGenTxs = []json.RawMessage{appGenTx}
-	}
-
-	appState, err := appInit.AppGenState(cdc, appGenTxs)
-	if err != nil {
-		return
-	}
-
-	err = writeGenesisFile(cdc, genFile, initConfig.ChainID, validators, appState)
-	if err != nil {
-		return
-	}
-
-	return
-}
-
-// append a genesis-piece
-func processGenTxs(genTxsDir string, cdc *codec.Codec) (
-	validators []types.GenesisValidator, appGenTxs []json.RawMessage, persistentPeers string, err error) {
-
-	var fos []os.FileInfo
-	fos, err = ioutil.ReadDir(genTxsDir)
-	if err != nil {
-		return
-	}
-
-	genTxs := make(map[string]server.GenesisTx)
-	var nodeIDs []string
-	for _, fo := range fos {
-		filename := path.Join(genTxsDir, fo.Name())
-		if !fo.IsDir() && (path.Ext(filename) != ".json") {
-			continue
+		for i, stdTx := range appGenTxs {
+			var jsonRawTx json.RawMessage
+			jsonRawTx, err = cdc.MarshalJSON(stdTx)
+			if err != nil {
+				return
+			}
+			genTxs[i] = jsonRawTx
 		}
 
-		// get the genTx
-		var bz []byte
-		bz, err = ioutil.ReadFile(filename)
+		appState, err = app.GaiaAppGenStateJSON(cdc, genTxs)
 		if err != nil {
 			return
 		}
-		var genTx server.GenesisTx
-		err = cdc.UnmarshalJSON(bz, &genTx)
+	} else {
```

Comment:
block comment on the alternative logic, which just creates a simple genesis file with no gentxs